pkg/network/http/telemetry.go

```diff
@@ -44,41 +64,32 @@ func (t *telemetry) aggregate(txs []httpTX, err error) {
 	}
 }
 
-func (t *telemetry) reset() telemetry {
-	now := time.Now()
-	then := atomic.SwapInt64(&t.then, now.Unix())
-
-	delta := telemetry{
-		misses:       atomic.SwapInt64(&t.misses, 0),
-		dropped:      atomic.SwapInt64(&t.dropped, 0),
-		rejected:     atomic.SwapInt64(&t.rejected, 0),
-		aggregations: atomic.SwapInt64(&t.aggregations, 0),
-		elapsed:      now.Unix() - then,
-	}
-
-	for i := range t.hits {
-		delta.hits[i] = atomic.SwapInt64(&t.hits[i], 0)
-	}
-
-	return delta
+func (t *telemetry) reset() {
+	atomic.SwapInt64(&t.aggregations, 0)
 }
 
-func (t *telemetry) report() {
-	var totalRequests int64
-	for _, n := range t.hits {
-		totalRequests += n
-	}
+func (t *telemetry) report() map[string]interface{} {
+	totalRequests := t.hits1XX + t.hits2XX + t.hits3XX + t.hits4XX + t.hits5XX
+	elapsed := time.Now().Unix() - t.then
+
+	stats := t.reporter.Report()
+	misses := stats["misses"].(int64)
+	dropped := stats["dropped"].(int64)
+	rejected := stats["rejected"].(int64)
+	aggregations := stats["aggregations"].(int64)
 
 	log.Debugf(
 		"http stats summary: requests_processed=%d(%.2f/s) requests_missed=%d(%.2f/s) requests_dropped=%d(%.2f/s) requests_rejected=%d(%.2f/s) aggregations=%d",
 		totalRequests,
-		float64(totalRequests)/float64(t.elapsed),
-		t.misses,
-		float64(t.misses)/float64(t.elapsed),
-		t.dropped,
-		float64(t.dropped)/float64(t.elapsed),
-		t.rejected,
-		float64(t.rejected)/float64(t.elapsed),
-		t.aggregations,
+		float64(totalRequests)/float64(elapsed),
+		misses,
+		float64(misses)/float64(elapsed),
+		dropped,
+		float64(rejected)/float64(elapsed),
+		rejected,
+		float64(dropped)/float64(elapsed),
+		aggregations,
 	)
+
```

Comment:
For this function, it seems like we should check the log level first.
The `misses`, `dropped`, etc. variables are assigned, and then the 4 floating point conversions and divisions are done regardless of log level.  if the entire function below `if stats:= .....` was `if loglevel >= debug`, then all of those computations wouldn't have to be done on every execution.